src/views/OrderDetailUpdated.vue

```diff
@@ -569,52 +569,43 @@ export default defineComponent({
+      const itemsToReject: any[] = [];
+      
       for (const item of order.part.items) {
-        let params = {} as any;
-        if(this.isEntierOrderRejectionEnabled(order)) {
-          params = {
-            ...payload,
-            rejectReason: item.rejectReason || this.rejectEntireOrderReasonId,
-            facilityId: item.facilityId,
+        const shouldReject = isEntireOrderRejection || item.rejectReason;
+
+        if (shouldReject) {
+          itemsToReject.push({
             orderItemSeqId: item.orderItemSeqId,
-            shipmentMethodTypeId: order.part.shipmentMethodEnum.shipmentMethodEnumId,
             quantity: parseInt(item.quantity),
-            ...(order.part.shipmentMethodEnum.shipmentMethodEnumId === "STOREPICKUP" && ({"naFacilityId": "PICKUP_REJECTED"})),
-          }
-        } else if(item.rejectReason) {
-          params = {
-            ...payload,
-            rejectReason: item.rejectReason || this.rejectEntireOrderReasonId,
             facilityId: item.facilityId,
-            orderItemSeqId: item.orderItemSeqId,
-            shipmentMethodTypeId: order.part.shipmentMethodEnum.shipmentMethodEnumId,
-            quantity: parseInt(item.quantity),
-            ...(order.part.shipmentMethodEnum.shipmentMethodEnumId === "STOREPICKUP" && ({"naFacilityId": "PICKUP_REJECTED"})),
-          }
+            updateQOH: false, // Could be true if QOH needs to be updated on rejection
+            rejectionReasonId: item.rejectReason || this.rejectEntireOrderReasonId,
+            kitComponents: isKit(item) ? item.rejectedComponents || [] : []
+          });
         }
-
-        if(Object.keys(params).length) {
-          // If the item is a kit, then pass the rejectedComponents with the item on rejection
-          if(isKit(item)) {
-            params["rejectedComponents"] = item.rejectedComponents
-          }
-          try {
-            const resp = await OrderService.rejectOrderItem({ payload: params });
-  
-            if(!hasError(resp)) {
-              order["part"] = {
-                ...order.part,
-                items: order.part.items.filter((orderItem: any) => !(orderItem.orderItemSeqId === item.orderItemSeqId && orderItem.productId === item.productId))
-              }
-            }
-          } catch(err) {
-            logger.error(`Something went wrong while rejecting order item ${item.productId}/${item.orderItemSeqId}`)
+      }
+      if (itemsToReject.length > 0) {
+        const payload = {
+          orderId: order.orderId,
+          rejectToFacilityId,
+          items: itemsToReject,
+          notify: true
+        };
+        try {
+          const resp = await OrderService.rejectOrderItems({ payload });
+
+          if (!hasError(resp)) {
+            // Remove rejected items from the part.items
+            const rejectedSeqIds = new Set(itemsToReject.map(i => i.orderItemSeqId));
```

Comment:
We need to check for the rejectedOrderItems list that we got in the response, as there might be case where any of the order item fails to reject.

Reply:
The service method `rejectOrderItems` invokes the `rejectOrderItem` service for each item within the same transaction. This means that if any single item fails to be rejected due to any reason, then the entire transaction will roll back. Which leads to all other items that were set to be rejected in that transaction will also remain in their original state and not be processed.